### PR TITLE
add best-of-five

### PR DESCRIPTION
**Pull request by alnkpa**

### CODE REVIEW

File: game_states.py [hunk 1]
```diff
@@ -90,13 +84,34 @@ def _should_proceed(self):
...
 
     def render(self):
         # maybe a fancy countdown animation at each tick?
         for paddle in self.game.players.values():
             self.game.screen.blit(paddle.surface, paddle.position)
 
+
+class StartingState(PauseState):
+    """
+    In this state, the game renders only the players' paddles for a given
```
Comment (hrantzsch) [inline game_states.py:105]: state comments should not be copy pasted ;)

File: game_states.py [hunk 2]
```diff
@@ -113,13 +128,42 @@ def render(self):
...
 
+    def __init__(self, game):
+        super().__init__(game)
+
     def update(self):
+        if self.game.ball.position.x <= 0:
+            uid = [p for p in self.game.players.values() if p.is_first_player][0].player_id
+            self.game.scores[uid] += 1
+        elif self.game.ball.position.x >= self.game.screen.width - 1:
+            uid = [p for p in self.game.players.values() if not p.is_first_player][0].player_id
```
Comment (hrantzsch) [inline game_states.py]: extract some functions for readability

File: game_states.py [hunk 3]
```diff
@@ -113,13 +130,68 @@ def render(self):
...
+        self.screen_center = Point(
+            self.game.screen.width // 2,
+            self.game.screen.height // 2,
+        )
+
+    def render(self):
+        super().render()
+
+        # get time we still have to wait and render it to center of screen
+        text = ':'.join(map(lambda i : str(i), self.game.scores.values()))
```
Comment (Bartzi) [inline game_states.py]: you can also write `map(str, self.game.scores.values())`

File: game_states.py [hunk 4]
```diff
@@ -113,13 +130,68 @@ def render(self):
...
+        self._check_if_scored()
+
         if self.game.ball.position.x <= 0 or \
-           self.game.ball.position.x >= self.game.screen.width-1:
-            self.game.state = GameOverState(self.game)
-            return
+         self.game.ball.position.x >= self.game.screen.width - 1:
+            if any([s > 2 for s in self.game.scores.values()]):
+                self.game.state = GameOverState(self.game)
+                return
```
Comment (Bartzi) [inline game_states.py:187]: only one return after the if, is sufficient I think ;)

**Review by Bartzi — CHANGES_REQUESTED** [review]
Please change the code according to the comments ;) and we should be good to go^^

File: game_states.py [hunk 5]
```diff
@@ -113,13 +131,64 @@ def render(self):
...
+
+        self.delay = 3  # seconds between inner state changes
+
+        self.next_state = RunningState(self.game)
+
+    def render(self):
+        super().render()
+
+        # get score and render it to center of screen
+        text = ':'.join(map(lambda i : str(i), self.game.scores.values()))
```
Comment (Bartzi) [inline game_states.py]: `map(str, self.game.scores.values())` is enough here ;)

**Pull request merged**